pandas/tests/frame/methods/test_to_csv.py

```diff
@@ -1285,3 +1285,33 @@ def test_to_csv_na_quoting(self):
         )
         expected = '""\n""\n'
         assert result == expected
+
+    def test_to_csv_categorical_and_ea(self):
+        # GH#46812
+        df = DataFrame({"a": "x", "b": [1, pd.NA]})
+        df["b"] = df["b"].astype("Int16")
+        df["b"] = df["b"].astype("category")
+        result = df.to_csv()
+        expected_rows = [",a,b", "0,x,1", "1,x,"]
+        expected = tm.convert_rows_list_to_csv_str(expected_rows)
+        assert result == expected
+
+    def test_to_csv_categorical_and_interval(self):
+        # GH#46297
+        with tm.assert_produces_warning(FutureWarning, check_stacklevel=False):
```

Comment:
if we `assert_produces_warning` we will need to remove this on the backport. (which is fine imo)

Reply:
sure

Reply:
Done